components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -128,6 +128,23 @@ pub fn year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>>
     Ok(Some(Int::from(t.year())))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn day_name(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Bytes>> {
+    match t {
+        Some(t) => {
+            if t.invalid_zero() {
+                return ctx
+                    .handle_invalid_time_error(Error::incorrect_datetime_value(&format!("{}", t)))
```

Comment:
I think `Error::incorrect_datetime_value(t)` is just enough, it takes `impl Display` as parameter type

Reply:
Indeed, I have modified.